exp/controllers/awsmanagedmachinepool_controller.go

```diff
@@ -266,3 +325,187 @@ func managedControlPlaneToManagedMachinePoolMapFunc(c client.Client, gvk schema.
 		return results
 	}
 }
+
+func (r *AWSManagedMachinePoolReconciler) getEC2Service(scope scope.EC2Scope) services.EC2MachineInterface {
+	return ec2.NewService(scope)
+}
+
+func (r *AWSManagedMachinePoolReconciler) reconcileLaunchTemplate(machinePoolScope *scope.ManagedMachinePoolScope, ec2Scope scope.EC2Scope) error {
```

Comment:
Is there anyway we could have a common function for this across [AWSMachinePool](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/833ca27c87eab8a8f7fd98ca85b46dcd1a22e94f/exp/controllers/awsmachinepool_controller.go#L383) and AWSManagedMachinePool?

Reply:
Do we need to extend the ec2 service? Or perhaps refactor and create a launch template service?

Reply:
Sounds good. I moved the logic into the launchtemplate service (`ReconcileLaunchTemplate`).